src/components/application_manager/src/commands/mobile/on_button_event_notification.cc

```diff
@@ -123,18 +125,8 @@ void OnButtonEventNotification::Run() {
 }
 
 void OnButtonEventNotification::SendButtonEvent(ApplicationConstSharedPtr app) {
-  if (!app) {
-    SDL_ERROR("OnButtonEvent NULL pointer");
-    return;
-  }
-
   smart_objects::SmartObjectSPtr on_btn_event =
-      new smart_objects::SmartObject();
-
-  if (!on_btn_event) {
-    SDL_ERROR("OnButtonEvent NULL pointer");
-    return;
-  }
+      utils::MakeShared<smart_objects::SmartObject>();
```

Comment:
@Kozoriz Using `MakeShared` avoids exception on `SmartObject` memory allocation, but `on_btn_event` could still be empty. So the check is needed.